### PR TITLE
Add Fortran module for GEFS data source in CDEPS

### PR DESCRIPTION
**Pull request by binli2337**
### Description of changes
The Fortran module for GEFS data source will be added to CDEPS.

### Specific notes
The revised code will include the following changes:
1) The Fortran module for GEFS "datm_datamode_gefs_mod.F90" will be added.
2) The nuopc cap code for data atmosphere model "atm_comp_nuopc.F90" will be updated.
3) CMakeLists.txt file will be updated to compile the updated code using cmake.

Testing performed:
Tests have been done in the mom6_cice6_cdeps application using a revised CMEPS code from ESCOMP/CMEPS (#74f7751, March 5, 2021 version).

### CODE REVIEW

**Comment by binli2337** [comment]
A related PR for ufs-weather-model will be submitted.

**Review by junwang-noaa — APPROVED** [review]

File: datm/datm_datamode_gefs_mod.F90 [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+!       end if
+
+       !--- temperature ---
+       if (tbotmax < 50.0_r8) Sa_tbot(n) = Sa_tbot(n) + tkFrz
+       ! Limit very cold forcing to 180K
+       Sa_tbot(n) = max(180._r8, Sa_tbot(n))
+!       Sa_ptem(n) = Sa_tbot(n)
+    end do
+
+    !----------------------------------------------------------
```
Comment (uturuncoglu) [inline datm/datm_datamode_gefs_mod.F90:227]: If there is no need to unit conversion. This part of code can be cleaned.

**Review by uturuncoglu — CHANGES_REQUESTED** [review]
I could not see CDEPS CFSR specific module file in here. Maybe you forgot to add it to PR.

**Comment by uturuncoglu** [comment]
> I could not see CDEPS CFSR specific module file in here. Maybe you forgot to add it to PR.

Sorry, that was already merged in previous PR.

**Pull request merged**